### PR TITLE
[OPPRO-301] Add expand operator

### PR DESCRIPTION
**Pull request by JkSelf**
## What changes were proposed in this pull request?

Add expand operator support in gluten with velox backend.

## How was this patch tested?
Added UT and TPC-DS verification

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-core/src/main/java/io/glutenproject/substrait/rel/ExpandRelNode.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+  public ExpandRelNode(RelNode input, String groupName,
+                       ArrayList<ArrayList<ExpressionNode>> groupings,
+                       ArrayList<ExpressionNode> aggExpressions,
+                       AdvancedExtensionNode extensionNode) {
+    this.input = input;
+    this.groupName = groupName;
+    this.groupings.addAll(groupings);
+    this.aggExpressions.addAll(aggExpressions);
+    this.extensionNode = extensionNode;
```
Comment (rui-mo) [inline gluten-core/src/main/java/io/glutenproject/substrait/rel/ExpandRelNode.java:46]: Looks like this Rel is similar as AggregateRel. Is there any special reason for not reusing the AggregateRel?

**Review by rui-mo — APPROVED** [review]
LGTM.

File: gluten-core/src/main/java/io/glutenproject/substrait/rel/ExpandRelNode.java (hunk 1 above)
Comment (JkSelf) [reply gluten-core/src/main/java/io/glutenproject/substrait/rel/ExpandRelNode.java:46]: It is not same with AggregateRel, which need Aggregate expressions not Aggregate Functions.

**Comment by JkSelf** [comment]
@rui-mo 
Will open other PRs if you have further comments. Thanks.

**Pull request merged**